### DISCUSSION TITLE
Error request to Spring-boot-Rsocket (Auth JWT)

### DISCUSSION THREAD

**Discussion by velesUst** [rsocket/rsocket-js#228] [Q&A] [answered]
Hello,  i have problem with authorization  from web-client.
The full description: 
https://stackoverflow.com/questions/72328729/error-connect-to-spring-boot-rsocket-auth-jwt-from-web-client-rsocketwebsocket

The server side is from tis examle.
https://github.com/feuyeux/kio

As i suppose,  the variants with metadata is not applicable in this case. I will be glad for any suggestions.

**Comment by viglucci** [comment, marked as answer]
Hi @velesUst ,

You are likely going to want to work with composite metadata and set your `metadataMimeType` to `MESSAGE_RSOCKET_COMPOSITE_METADATA.string`.

The important bit is going to be the routing metadata, which is what tells the server how to route the incoming RSocket request.

I haven't dug through the server example code you linked on StackOverflow, but just by looking at your example code, you would supply the routing metadata with your `requestStream` as so:

Also, the [example project](https://github.com/feuyeux/kio#design) you listed though references `signin` as a `request/response` so you actually don't want `requestStream`, but `requestResponse`.

```js
socket
  .requestResponse({
    data: Buffer.from(JSON.stringify({
      user_id: '0000',
      password: 'Zero4'
    })),
    metadata: encodeCompositeMetadata([
      [MESSAGE_RSOCKET_ROUTING, encodeRoute("signin.v1")],
    ]),
  })
```

You will likely want to use `BufferEncoders`, as shown in this [example](https://github.com/rsocket/rsocket-js/blob/cdfa824513ae58c5ef8abb14c3754c99199296b8/packages/rsocket-examples/src/CompositeMetadataExample.js). And additionally, I believe you should not use `JsonSerializer` for the metadata, but instead `IdentitySerializer`, which will pass the composite metadata buffer straight through, rather than trying to serialize to and from JSON.

You may still run into some issues, but I suspect that this will get you past the `metadata is malformed ERROR` error.

Hope that helps.

**Reply by velesUst** [reply]
Grate thanks for the detailed advices! All works according to directions.
